File: python/src/patterns/creational/builder/query_client.py
from src.patterns.creational.builder.query import Query

def main():
    query = (
        Query.Builder("users", "id", "name", "email")
        .where("age > 18")
        .order_by("name")
        .limit(10)
        .offset(0)
        .build()
    )
    print(query)
    print(query.to_sql())

if __name__ == "__main__":
    main()

File: java/src/patterns/creational/builder/QueryClient.java
package patterns.creational.builder;

public class QueryClient {
    public static void main(String[] args) {
        Query query = new Query.Builder("users", "id", "name", "email")
            .where("age > 18")
            .orderBy("name")
            .limit(10)
            .offset(0)
            .build();
        System.out.println(query);
        System.out.println(query.toSql());
    }
}